app/controllers/concerns/web_app_controller_concern.rb

```diff
@@ -45,6 +46,12 @@ def redirect_unauthenticated_to_permalinks!
 
   protected
 
+  def redirect_to_tos_interstitial!
+    return unless current_user&.require_tos_interstitial
+
+    redirect_to(terms_of_service_interstitial_url)
```

Comment:
Could we just render it here instead of creating an addressable page?

Reply:
I don't want to create a referenceable page for the interstitial, also because its naming will be awkward. Willing to get a third opinion from @renchap.